server/datastore/mysql/migrations/tables/20220704101843_UpdateSoftwareIDInSoftwareCVE.go

```diff
@@ -0,0 +1,62 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220704101843, Down_20220704101843)
+}
+
+func Up_20220704101843(tx *sql.Tx) error {
+	var min int
+	var max int
+
+	const selectStmt = `
+SELECT COALESCE(MIN(cve.id), 0) AS min_id, COALESCE(MAX(cve.id), 0) as max_id 
+FROM software_cve AS cve
+WHERE cve.software_id IS NULL;`
+	if err := tx.QueryRow(selectStmt).Scan(&min, &max); err != nil {
+		return errors.Wrap(err, "selecting min,max id")
+	}
+
+	// Update in batches
+	const batchSize = 500
```

Comment:
+1 on adding logging - I'll get it done.

Yup it should be idempotent, because new rows should have the `software_id` column populated.

Reply:
> And if this is running while fleet is running, the worse that could happen is that a row would get updated twice because of the raise... (thinking out loud here).
> 
:thinking: not sure how we can get to the scenario where a row is updated twice - `fleet` doesn't do updates to the `software_cve` table, only selects, inserts and deletes.

> Could we load main up in loadtest, throw a ton of hosts at it, and then try running this migration while everything is active and see if we have our first live migration? :)
>
Sure, but I'm not 100% clear how we could load test this - unless I'm missing something, by the time the hosts are added the migrations are already done running, no?

Reply:
@chiiph 
1. First DDL statement (adding the new column):
![ddl_1](https://user-images.githubusercontent.com/8041318/177678832-b5f34cfc-50b5-4d5f-bdf4-cf089bc7c7a5.png)

2. Updating `software_id` column (I only did 100 iterations of the update loop):
![dml_1_1](https://user-images.githubusercontent.com/8041318/177678900-c4cc0a78-3baa-40d8-9c02-edf2b87dcdfe.png)
![dml_1_2](https://user-images.githubusercontent.com/8041318/177678909-440ee520-5bed-4ca7-8700-e676852341e1.png)

3. Last DDL statement (adding the index): 
![ddl_2](https://user-images.githubusercontent.com/8041318/177679022-ce632d99-afa3-4731-950c-c2ab194a8646.png)

Reply:
Agreed it looks good!